pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -605,6 +610,58 @@ func (b *BridgePodInterface) createBridge() error {
 	return nil
 }
 
+func (b *BridgePodInterface) switchPodInterfaceWithDummy() error {
+	originalPodInterfaceName := b.podInterfaceName
+	newPodInterfaceName := fmt.Sprintf("%s-nic", originalPodInterfaceName)
+	dummy := &netlink.Dummy{LinkAttrs: netlink.LinkAttrs{Name: originalPodInterfaceName}}
+
+	if b.podNicLink.Type() != "dummy" {
+		// Bring down link to rename it
+		if err := Handler.LinkSetDown(b.podNicLink); err != nil {
```

Comment:
Since `preparePodNetworkInterfaces` is setting the link to down and then up anyway. I would remove the setting to down and up from `switchPodInterfaceWithDummy` and just move setting the link to down to the beginning of `preparePodNetworkInterfaces` instead of setting it to down and up twice.

Reply:
agreed